### ISSUE TITLE
Allow using special characters in names of custom units (ř,č,í...)

### ISSUE THREAD

**Issue by danbulant** [josdejong/mathjs#1663]
Hello.
I want to use mathjs in Czech platform, and I would like to have support for czech. How I can translate? I've found only createUnit, but this only fixes units, but I would like to change even the word "to" and maybe other words that are available.
Also using special characters that normally appear in non-english words are throwing errors.

**Comment by danbulant** [comment]
If it's not possible in API, could you please send me location of the strings in source code?

**Comment by josdejong** [comment]
I think you're asking for two things here:

1. Can I get localized error messages? (the error messages is the only text in mathjs I think)

    There is no localization implemented in mathjs right now. That will be useful to implement I think. Anyone interested in implementing localization in mathjs?

2. I cannot use certain special characters in variable names in the expression parser. Is this what you mean? Can you give an example of what you try to do?

**Comment by josdejong** [comment]
> If it's not possible in API, could you please send me location of the strings in source code?

Check out the source code. Search for all occurrences of `"throw "` in the `src` folder, you will find about 700 error messages.

**Comment by m93a** [comment]
I don't have time to implement error localization right now, but once somebody does it I can help with translation to Czech 😉

**Comment by josdejong** [comment]
Thanks for your offer already :smile: :+1:

**Comment by danbulant** [comment]
I didn't mean errors - they're for devs and such its OK to use English only.

The problem is when using units in calculations. 
You can use custom named, but those custom named don't accept these characters and will throw an error.

**Comment by josdejong** [comment]
Hm, now I'm confused a bit, so besides translation you would like to be able to use special characters in names of custom units?

**Comment by danbulant** [comment]
I don't need translation of errors (I just wrongly described my problem).The actual problem is that when I try using custom units, such as vteřiny (seconds), it throws error because ř isn't in the regex check. If I removed the regex check, the code fails with another error... On Feb 26, 2020 9:29 AM, Jos de Jong <notifications@github.com> wrote:Hm, now I'm confused a bit, so besides translation you would like to be able to use special characters in names of custom units?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
Sent via Migadu.com, world's easiest email hosting>

**Comment by josdejong** [comment]
Ok in that case I will update the title of this issue ok?

**Comment by rnd-debug** [comment]
@josdejong Looks related to those [checks](https://github.com/josdejong/mathjs/blob/23f32fae9c18a6b27cffc25c68f5304e1cf66854/src/type/unit/Unit.js#L220-L232)
I see two approaches to allow the usage of "non-roman" alphabetical chars:
- either a hack:
`myChar.toUpperCase() != myChar.toLowerCase()` => would be true for languages where the uppercase letter is different from the lowercase.
- or a more standard:
`myChar.match(/^\p{Letter}/u)`  =>uses [Unicode property escapes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Unicode_Property_Escapes) which might not be supported on [old browsers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Browser_compatibility)

What would be preferable?

**Comment by josdejong** [comment]
We could also do something similar to `parse.js`, where a function `isAlpha` is used which can be overridden if you need more characters: https://github.com/josdejong/mathjs/blob/develop/src/expression/parse.js#L445-L467. Maybe we can unify this, not sure.

**Comment by rnd-debug** [comment]
@josdejong How about that #2000 ? (Happy 2000-th PR, btw :) )
Usage case snippet:
```
const { create, all } = require("./dist/math");
const config = {}
const math = create(all, config);

const isAlphaOriginal = math.Unit.isValidAlpha
math.Unit.isValidAlpha = function (c) {
  return isAlphaOriginal(c) 
  || ((c).toUpperCase() !== (c).toLowerCase())
}

math.createUnit('vteřiny', '1 s')
console.log(math.evaluate('1vteřiny + 1mins').toNumber('s'))   // returns 61
```

**Comment by josdejong** [comment]
Thanks, your PR looks good, would be great if you can add tests and documentation for it

> (Happy 2000-th PR, btw :) )

😂 thanks to you 👍

**Comment by josdejong** [comment]
The new function `Unit.isValidAlpha` is now available in `v7.6.0`, thanks @rnd-debug
